src/megachatapi_impl.cpp

```diff
@@ -1792,6 +1792,46 @@ void MegaChatApiImpl::sendPendingRequests()
             break;
         }
 
+        case MegaChatRequest::TYPE_SET_RETENTION_TIME:
+        {
+            MegaChatHandle chatid = request->getChatHandle();
+            int period = request->getParamType();
+            bool inSeconds = request->getFlag();
+
+            if (chatid == MEGACHAT_INVALID_HANDLE)
+            {
+                errorCode = MegaChatError::ERROR_ARGS;
+                break;
+            }
+
+            ChatRoom *chatroom = findChatRoom(chatid);
```

Comment:
I think we have to check if period is greater or equal to zero with
I we don't check that, we have to prepare to handle retention time negative

Reply:
To disable retention time you can provide zero or negative number.
I forgot to update the documentation, sorry about that. I've changed at
1c55e12

Reply:
We are handling the value received from chatd as `unsigned int`, if we set a negative value, chatd returns 0 or negative?

Reply:
Chatd returns 0

Reply:
Then, it's perfect